packages/zipkin/src/batch-recorder.js

```diff
@@ -135,6 +147,15 @@ class BatchRecorder {
     }
   }
 
+  flush() {
+    this.partialSpans.forEach((span, id) => {
+      if (span.delegate.duration === undefined) {
```

Comment:
if the span is finished, we don't want to add akward annotations outside the span lifespan. Do you agree?